ivadomed/transforms.py

```diff
@@ -258,26 +259,72 @@ def __call__(self, sample, metadata=None):
         is_2d = sample.shape[-1] == 1
         metadata['preresample_shape'] = sample.shape
         zooms = list(metadata["zooms"])
-
         if len(zooms) == 2:
             zooms += [1.0]
-
-        hfactor = zooms[0] / self.hspace
-        wfactor = zooms[1] / self.wspace
-        dfactor = zooms[2] / self.dspace
+        if self.flag_pixel:
+            hfactor = self.hspace / sample.shape[0]
+            wfactor = self.wspace / sample.shape[1]
+            dfactor = self.dspace / sample.shape[2]
+        else:     
+            hfactor = zooms[0] / self.hspace
+            wfactor = zooms[1] / self.wspace
+            dfactor = zooms[2] / self.dspace
         params_resample = (hfactor, wfactor, dfactor) if not is_2d else (hfactor, wfactor, 1.0)
 
         # Run resampling
         data_out = zoom(sample,
                         zoom=params_resample,
                         order=1 if metadata['data_type'] == 'gt' else 2)
-
```

Comment:
Please revert this change. There should be no change here.

Reply:
fixed.